scipy/stats/tests/test_kdeoth.py

```diff
@@ -318,6 +318,51 @@ def test_kde_integer_input():
     assert_array_almost_equal(kde(x1), y_expected, decimal=6)
 
 
+@pytest.mark.parametrize("bw_type", [getattr(np, dtype, None) for dtype in
```

Comment:
The code in these decorators could be simplified by determining the floating point types once,  and filtering out the undefined types instead of substituting `None`, i.e.:

```
_ftypes = [getattr(np, dtype) for dtype in ['float32', 'float64',
                                            'float96', 'float128']
           if hasattr(np, dtype)]
```

Then use `_ftypes` in the decorators, instead of repeating the list comprehension in each one.

Reply:
Yes, thats nicer!